conda_forge_tick/migrators/replacement.py

```diff
@@ -147,3 +149,54 @@ def migrator_uid(self, attrs: "AttrsTypedDict") -> "MigrationUidTypedDict":
         n = super().migrator_uid(attrs)
         n["name"] = self.name
         return n
+
+
+class MiniReplacement(MiniMigrator):
+    """Minimigrator for replacing one package with another.
+
+    Parameters
+    ----------
+    old_pkg : str
+        The package to be replaced.
+    new_pkg : str
+        The package to replace the `old_pkg`.
+    """
+
+    allowed_schema_versions = [0, 1]
+
+    def __init__(
+        self,
+        *,
+        old_pkg: "PackageName",
+        new_pkg: "PackageName",
+        requirement_types: tuple[str] = ("host",),
```

Comment:
Oh I am sorry I was mistaken. The old "xz" one did everything in the text file as long as it matched ` xz\n`. So I think actually what we want is none of the above and something much more general. We should inspect the other migrators to see what they look like too.

Reply:
Well, yes and no. It replaced everything but `filter()`-ed on `host` requirements. In that sense, the new minimigrator does the same thing. Ofc, I'm open to changing the behavior to something "saner".

Reply:
Right. My concern is us missing things like ignored run exports in `build/ignore_run_exports_from`, etc. The current migrator would catch that and we need to replicate that kind of behavior. I don't have an exhaustive list of where say `xz` might appear, which is why something more generic would be better.

Reply:
Just the word matching.